src/client/dfs/duns.c

```diff
@@ -375,9 +386,11 @@ duns_resolve_path(const char *path, struct duns_attr_t *attr)
 		/** parse the pool uuid */
 		rc = uuid_parse(t, attr->da_puuid);
 		if (rc) {
-			D_ERROR("Invalid format: pool UUID cannot be parsed\n");
-			D_FREE(dir);
-			return EINVAL;
+			attr->da_pool_label = strdup(t);
```

Comment:
no because this is for external use. same comment applies for the cont ones, and the associated free

Reply:
This is in a shared library which could be used externally or by our tools.  We need to be consistent or our tools for leak detection won't work.

Reply:
Im not sure you understood what im saying.
the da_pool/cont_label is for external use, allocated by the library but should be freed by the user. the DAOS tools is one user here. so we can't use the macros as it would cause the alloc being recorded and the free never seen before external users will not use D_FREE.

Reply:
We already have leaks in the new go based daos command because it's not freeing strings allocated in this function, I wonder if there would be a case for a "destroy" function that correctly released all this memory.